cdap-common/src/main/java/co/cask/cdap/common/conf/PluginClassDeserializer.java

```diff
@@ -57,7 +57,7 @@ public PluginClass deserialize(JsonElement json, Type typeOfT,
       ? context.<Map<String, PluginPropertyField>>deserialize(jsonObj.get("properties"), PROPERTIES_TYPE)
       : ImmutableMap.<String, PluginPropertyField>of();
 
-    return new PluginClass(type, name, description, className, null, properties);
+    return new PluginClass(type, name, description, className, null, properties, null);
```

Comment:
This should try to deserialize the methods and pass it in.  At the very least it should pass in an empty object instead of null